### PR TITLE
Fix item descriptions

### LINKED ISSUE #273: Dodoco description text error
image

### PR DESCRIPTION
**Pull request by vapopescu**
Fixes #273
These changes instruct the pipeline to substitute `\n` to `<br/>` in the weapon and material descriptions. Also fixed some code-duplication in 'WeaponSheet.tsx'

Resulting Changes:
- 'Dodoco Tales' has correct item description
- 'Sword of Descension' and 'Predator' (PSN Traveler and Aloy weapons) have correct passive description
- Material descriptions are HTML-safe, for future-proofing

image
image
image

### CODE REVIEW

File: apps/frontend/src/app/Data/Weapons/WeaponSheet.tsx [hunk 1]
```diff
@@ -33,17 +34,14 @@ export default class WeaponSheet {
...
-  static tr(key: string) {
-    return (strKey: string) => (
-      <Translate
-        ns={key === 'QuantumCatalyst' ? `weapon_${key}` : `weapon_${key}_gen`}
-        key18={strKey}
-      />
-    )
+  static tr(key: WeaponKey) {
+    return trans('weapon', key)[0]
   }
```
Comment (vapopescu) [inline apps/frontend/src/app/Data/Weapons/WeaponSheet.tsx:43]: There may be a more efficient way to write this, but I'm not that good with TypeScript

File: libs/dm-localization/src/lib/parseUtil.ts [hunk 2]
```diff
@@ -77,6 +77,16 @@ const paragraph = (string) => {
   while (parsed[parsed.length - 1] === '<br/>') parsed.pop()
   return { ...parseBulletPoints(parsed) }
 }
+
+/**
+ * Substitutes escaped newlines ('\\n') with linebreaks ('<br/>')
+ * @param string the string to parse
+ * @returns the resulting string
+ */
+const paragraph2 = (string: string) => {
```
Comment (vapopescu) [inline libs/dm-localization/src/lib/parseUtil.ts:86]: Function name may be changed. I was lazy :)

**Review by nguyentvan7 — APPROVED** [review]

**Comment by frzyc** [comment]
I think the proper fix is to amend the original `paragraph` function:
```
const parsed = string.split('\\n').map((s) => s || '<br/>')
```
to
```
const parsed = string.split(/\\{1,2}n/gm).map((s) => s || '<br/>')
```

**Review by frzyc — CHANGES_REQUESTED** [review]
I think the proper fix is to amend the original paragraph function:
```
const parsed = string.split('\\n').map((s) => s || '<br/>')
```
to
```
const parsed = string.split(/\\{1,2}n/gm).map((s) => s || '<br/>')
```

**Comment by vapopescu** [comment]
@frzyc Sorry for being a newb,. It seems thats, React is happy to receive an array as well as strings, so the original `paragraph` filter already worked as-is. It just needed to be used here. As a side-effect, weapon descriptions are now arrays, and weapon passive descriptions are 2D-arrays. E.g. Favonius codex

```
{
  "name": "Favonius Codex",
  "description": {
    "0": "A secret tome that belonged to the scholars of the Knights of Favonius. It describes the logic and power of elements and matter."
  },
  "passiveName": "Windfall",
  "passiveDescription": {
    "0": {
      "0": "CRIT Hits have a <cryo>60%</cryo> chance to generate a small amount of Elemental Particles, which will regenerate 6 Energy for the character. Can only occur once every <cryo>12</cryo>s."
    },
    "1": {
      "0": "CRIT Hits have a <cryo>70%</cryo> chance to generate a small amount of Elemental Particles, which will regenerate 6 Energy for the character. Can only occur once every <cryo>10.5</cryo>s."
    },
    "2": {
      "0": "CRIT Hits have a <cryo>80%</cryo> chance to generate a small amount of Elemental Particles, which will regenerate 6 Energy for the character. Can only occur once every <cryo>9</cryo>s."
    },
    "3": {
      "0": "CRIT Hits have a <cryo>90%</cryo> chance to generate a small amount of Elemental Particles, which will regenerate 6 Energy for the character. Can only occur once every <cryo>7.5</cryo>s."
    },
    "4": {
      "0": "CRIT Hits have a <cryo>100%</cryo> chance to generate a small amount of Elemental Particles, which will regenerate 6 Energy for the character. Can only occur once every <cryo>6</cryo>s."
    }
  }
}
```

**Review by frzyc — APPROVED** [review]

**Review by nguyentvan7 — APPROVED** [review]

**Pull request merged**